src/xml.rs

```diff
@@ -27,221 +33,429 @@ pub struct Document {
     ///
     /// The latest version is 1,0,0.
     #[serde(rename = "@Version")]
-    version: String,
+    pub version: String,
+    /// Product
     #[serde(rename = "PRODUCT")]
-    product: Product,
+    pub product: Product,
+    /// Collection of all tracks
     #[serde(rename = "COLLECTION")]
-    collection: Collection,
+    pub collection: Collection,
+    /// Playlist Tree
     #[serde(rename = "PLAYLISTS")]
-    playlists: Playlists,
+    pub playlists: Playlists,
 }
 
+/// Product
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
-struct Product {
+pub struct Product {
     /// Name of product
     ///
     /// This name will be displayed in each application software.
     #[serde(rename = "@Name")]
-    name: String,
+    pub name: String,
     /// Version of application
     #[serde(rename = "@Version")]
-    version: String,
+    pub version: String,
     /// Name of company
     #[serde(rename = "@Company")]
-    company: String,
+    pub company: String,
 }
 
 /// The information of the tracks who are not included in any playlist are unnecessary.
-#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
-struct Collection {
-    /// Number of TRACK in COLLECTION
-    #[serde(rename = "@Entries")]
-    entries: i32,
+#[derive(Debug, PartialEq, Clone, Deserialize)]
+pub struct Collection {
+    // The "Entries" attribute that contains the "Number of TRACK in COLLECTION" is omitted here,
+    // because we can just take the number of elements in the `tracks` vector instead.
+    /// Tracks in the collection.
     #[serde(rename = "TRACK")]
-    track: Vec<Track>,
+    pub tracks: Vec<Track>,
+}
+
+impl Serialize for Collection {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        #[derive(Serialize)]
+        struct Value<'a> {
+            /// Number of TRACK in COLLECTION
+            #[serde(rename = "@Count")]
+            entries: usize,
+            /// Tracks
+            #[serde(rename = "TRACK")]
+            tracks: &'a Vec<Track>,
+        }
+
+        let value = Value {
+            entries: self.tracks.len(),
+            tracks: &self.tracks,
+        };
+
+        value.serialize(serializer)
+    }
```

Comment:
I feel like this could be simplified by using `#[serde(getter = "tracks.len()")]` instead?